crates/biome_js_analyze/src/lint/nursery/no_useless_length_check.rs

```diff
@@ -0,0 +1,366 @@
+                return None;
+            };
+
+            let task_member_name_node = task_member_exp.member_name()?;
+            let task_member_name = task_member_name_node.text();
+            match expect_error {
+                ErrorType::UselessLengthCheckWithEvery => {
+                    if task_member_name == "every" {
+                        array_tokens_used_api.insert(task_target_token.text());
+                        Some(())
+                    } else {
+                        None
+                    }
+                }
+                ErrorType::UselessLengthCheckWithSome => {
+                    if task_member_name == "some" {
+                        array_tokens_used_api.insert(task_target_token.text());
+                        Some(())
+                    } else {
+                        None
+                    }
+                }
+            }
+        }
+        // ( foo )
+        AnyJsExpression::JsParenthesizedExpression(parent_exp) => search_logical_exp(
+            &parent_exp.expression().ok()?,
+            replacer,
+            expect_error,
+            comparing_zeros,
+            array_tokens_used_api,
+        ),
+        AnyJsExpression::JsIdentifierExpression(_) => Some(()),
+        _ => None,
+    }
+}
+
+fn get_parenthesized_parent(exp: AnyJsExpression) -> AnyJsExpression {
+    let Some(parent) = exp.syntax().parent() else {
+        return exp;
+    };
+    let Some(parent) = JsParenthesizedExpression::cast(parent) else {
+        return exp;
+    };
+    get_parenthesized_parent(AnyJsExpression::from(parent))
+}
+
+fn get_parenthesized_child(exp: &AnyJsExpression) -> AnyJsExpression {
+    let AnyJsExpression::JsParenthesizedExpression(parenthesized) = exp else {
+        return exp.clone();
+    };
+    let Some(child) = parenthesized.expression().ok() else {
+        return exp.clone();
+    };
+    get_parenthesized_child(&child)
+}
+
+impl Rule for NoUselessLengthCheck {
+    type Query = Ast<JsLogicalExpression>;
+    type State = (ErrorType, Replacer);
+    type Signals = Vec<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+
+        let Some(operator) = node.operator_token().ok() else {
+            return Vec::new();
+        };
+        // node must not be a child of a logical expression
+        if is_logical_exp_descendant(&AnyJsExpression::from(node.clone()), operator.kind()) {
+            return Vec::new();
+        }
+
+        let mut fixable_list: Vec<(ErrorType, Replacer)> = Vec::new();
+
+        for err_type in [
+            ErrorType::UselessLengthCheckWithEvery,
+            ErrorType::UselessLengthCheckWithSome,
+        ] {
+            let mut comparing_zeros: HashMap<String, Vec<Replacer>> = HashMap::new();
+            let mut array_tokens_used_api: HashSet<String> = HashSet::new();
```

Comment:
```suggestion
            let mut comparing_zeros = HashMap::new();
            let mut array_tokens_used_api = HashSet::new();
```

I believe the compiler is able to infer the correct type without casting. Usually, in Rust, you want to use the cast only when the compiler isn't able to infer it, and when it doesn't, it usually tells you with a compiler error.